timesketch/tsctl.py

```diff
@@ -645,12 +661,27 @@ def validate_context_links_conf(path):
         print("The provided config file is empty.")
         return
 
-    for entry in context_link_config:
-        try:
-            validate(instance=context_link_config[entry], schema=schema)
-            print(f'=> OK: "{entry}"')
-        except (ValidationError, SchemaError) as err:
-            print(f'=> ERROR: "{entry}" >> {err}\n')
+    if context_link_config["hardcoded_modules"]:
+        for entry in context_link_config["hardcoded_modules"]:
+            try:
+                validate(
+                    instance=context_link_config["hardcoded_modules"][entry],
```

Comment:
Can you explain what "hardcoded_modules" are and why we need them?

Reply:
See the explanation in the [context_links.yaml](https://github.com/google/timesketch/pull/2897/files#diff-269e304c1e3c351af85fde89dec9ea1c38dc42663291d29bf783e4c27ee1a19cR13) file. 

TL;DR: `hardcoded_moduled` are context links that are predefined in timesketch. E.g. unfurl and xml formatter. Having them as part of the configuration allows for TS operators to deactivate them if they don't want them and also easily define what fields they should trigger on.